test/app-tap/module_api.c

```diff
@@ -2335,6 +2335,80 @@ test_tuple_validate_formatted(lua_State *L)
 	return 1;
 }
 
+static int
+session_check_admin(lua_State *L)
+{
+	int sess_id = box_session_id();
+	bool sess_exist = box_session_exists(sess_id);
+	assert(sess_exist);
+
+	sess_id = box_session_id();
+	char peer_str_arg[100];
+	const char *peer_str = box_session_peer(sess_id, peer_str_arg, 100);
+    bool peer_is_null = (peer_str == NULL);
+	assert(peer_is_null);
+
+	bool sess_not_exist = !box_session_exists(1234567890);
+	assert(sess_not_exist);
+
+	lua_pushboolean(L, 1);
+
+	return 1;
+}
+
+static int
+fiber_session_check_func(va_list va)
+{
+	(void) va;
+	do {
+		fiber_set_cancellable(true);
+		fiber_sleep(0.01);
```

Comment:
Why do you need a sleep? Why can't the whole function just return the session ID and exit?

Reply:
The logic for launching background fiber with some work and waiting for it to exit is copied from here https://github.com/tarantool/tarantool/blob/master/test/app-tap/module_api.c#L258-L285

Reply:
I didn't review the patch yet, will do later, but I just warn you - the fact that something is done in some other place in one way does not mean you need to blindly copy it in the new code. Old code != good code in 100% cases. It is never a justification, not in a bit. Please, revisit the comments where you justify them as 'they are copied' bearing this in mind before I will do a next review.